ArduCopter/GCS_Mavlink.cpp

```diff
@@ -826,6 +827,31 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_packet(const mavlink_command_
         return MAV_RESULT_FAILED;
 #endif
 
+#if MODE_GUIDED_ENABLED == ENABLED
+     case MAV_CMD_NAV_LOITER_TURNS:
+        if (copter.flightmode->in_guided_mode())
+        {
+            Location circle_center = Location((int32_t)(packet.param5*1e7), (int32_t)(packet.param6*1e7), (int32_t)(packet.param7*100), copter.current_loc.get_alt_frame());
```

Comment:
```suggestion
            Location circle_center{
               (int32_t)(packet.param5*1e7),
               (int32_t)(packet.param6*1e7),
               (int32_t)(packet.param7*100),
               MAV_FRAME_GLOBAL_RELATIVE_ALT
           };
```

 - don't construct an object on the stack and assign from it into another object on the stack (compilers will often fix this for you)
 - formatting
 - use braces to get the compiler to warn you if you're narrowing variables
 - don't assume the command coming in is in the same frame Copter is using
 - that frame assumption should be checked!
 -

Reply:
Done! Thanks for such a detailed reason. Helped me learn new things